pandas/core/shared_docs.py

```diff
@@ -383,8 +383,11 @@
     "storage_options"
 ] = """storage_options : dict, optional
     Extra options that make sense for a particular storage connection, e.g.
-    host, port, username, password, etc., if using a URL that will
-    be parsed by ``fsspec``, e.g., starting "s3://", "gcs://". An error
-    will be raised if providing this argument with a non-fsspec URL.
-    See the fsspec and backend storage implementation docs for the set of
-    allowed keys and values."""
+    host, port, username, password, etc. If using a URL that will
```

Comment:
Maybe something a long the lines of
"[... host, port, username, password, etc.] Key-value pairs for HTTP(S) URLs are forwarded to``urllib`` as header options. For other URLs , e.g., starting with "s3://" and "gcs://", key-value pairs are forwarded to ``fsspec``. Please see ``fsspec`` and ``urllib`` for more details.

Reply:
Sounds reasonable enough to me.  Unrelated to the documentation change (but something I'll address in the next commit if you could provide guidance) I noticed that some of the parquet tests I wrote are failing on the checks that get run due to there not being a usable engine.  Am I missing something there or should I try to write tests that don't rely on having a parquet parsing engine?  Also, it looks like we don't like unittest and prefer pytest with monkeypatching, but I can't seem to find a MagicMock equivalent in pytest.  Would we rather have the import or should I write my own class for that?  Thanks for the guidance!

Reply:
About the docs, maybe even replace `fsspec` with `fsspec.open(**storage_options)` and `urllib` with `urllib.request.Request(headers=storage_options)` but that might be too verbose, idk.

Parquet needs optional dependencies, either `pyarrow` of `fastparquet`. It would be good to parametrize your test to cover both engines (if both engines support the new feature), then you can easily add a `pytest.importorskip`.

Sorry, I'm not familiar with monkeypatching/MagicMock.